File: sdk/unity/sdk/Assets/KingstonDynamics/DragonSong/Protocol/Utils/Prefixinator.cs
namespace KingstonDynamics.DragonSong.Protocol.Utils
{
    public static class Prefixinator
    {

        public static byte[] Prefix(byte[] data)
        {
            return Concatinator.ConctatinateByteArrays(Byteinator.IntToBytes(data.Length), data);
        }
    }
}

File: src/old/DragonSong-Protocol/src/main/java/com/kingstondynamics/dragonsong/protocol/utils/Prefixinator.java
package com.kingstondynamics.dragonsong.protocol.utils;

public class Prefixinator {
    
    /**
     * Private Constructor for Utility Class
     */
    private Prefixinator() {
        throw new AssertionError();
    }
    
    public static byte[] prefix(byte[] data) {
        return Concatinator.ConcatinateByteArrays(Byteinator.intToBytes(data.length), data);
    }
}
